mlos_bench/mlos_bench/storage/sql/experiment_data.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+An interface to access the benchmark data stored in SQL DB.
+"""
+
+from typing import Dict
+
+from sqlalchemy import Engine
+
+from mlos_bench.environments.status import Status
+from mlos_bench.storage.base_experiment_data import ExperimentData
+from mlos_bench.storage.sql.schema import DbSchema
+from mlos_bench.storage.base_trial_data import TrialData
+from mlos_bench.storage.sql.trial_data import TrialSqlData
+
+
+class ExperimentSqlData(ExperimentData):
+    """
+    Base interface for accessing the stored benchmark data.
+    """
+
+    def __init__(self, engine: Engine, schema: DbSchema, exp_id: str):
+        super().__init__(exp_id)
+        self._engine = engine
+        self._schema = schema
+
+    @property
+    def trials(self) -> Dict[int, TrialData]:
+        with self._engine.connect() as conn:
```

Comment:
Maybe not here, but we should fix this to use a connection pool to avoid repeated connections if possible.

Reply:
As far as I understand, SQLAlchemy uses a connection pool under teh hood - `engine.connect()` and `engine.begin()` just get a connection from that pool and wrap it in python context so it can be returned to the pool (and committed, in case of `.begin()`) on the context `__exit__()`

Reply:
Agree. I will add `experiment.all_trial_{results,telemetry}` in a moment

Reply:
added `experiment.results` that returns trial status, config, and results in one dataframe.